### ISSUE TITLE
[BUG] Update of HANA Cloud service instance fails

### ISSUE THREAD

**Issue by lechnerc77** [SAP/terraform-provider-btp#551]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of the Terraform provider are you using?

0.6.0-beta2

### What version of the Terraform CLI are you using?

1.5.7

### What type of issue are you facing

bug report

### Describe the bug

When updating a service instance of HANA Cloud (e.g. changing the memory) the following error occurs:

```bash
 Error: API Error Updating Resource Service Instance (Subaccount)
│ 
│   with btp_subaccount_service_instance.hana_cloud_hana_instance,
│   on main.tf line 25, in resource "btp_subaccount_service_instance" "hana_cloud_hana_instance":
│   25: resource "btp_subaccount_service_instance" "hana_cloud_hana_instance" {
│ 
│ json: cannot unmarshal object into Go struct field OperationResponseObject.last_operation.errors of type []servicemanager.ModelError
```

### Expected Behavior

Either a meaningful error is returned (maybe due to wrong parameter settings) or the service instance gets updated

### Steps To Reproduce

The following script can be used to create the initial version of the service instance (assumption: global account is entitled for "HANA Cloud"

```terraform
resource "btp_subaccount" "retest-hana-cloud" {
  name      = "retest-hana-cloud"
  subdomain = "retest-hana-cloud"
  region    = "eu12"
}


resource "btp_subaccount_entitlement" "hana-cloud" {
  subaccount_id = btp_subaccount.retest-hana-cloud.id
  service_name  = "hana-cloud"
  plan_name     = "hana"
  #amount        = 1
}

data "btp_subaccount_service_plan" "by_name" {
  subaccount_id = resource.btp_subaccount.retest-hana-cloud.id
  offering_name = "hana-cloud"
  name          = "hana"
  depends_on = [ btp_subaccount_entitlement.hana-cloud ]
}


resource "btp_subaccount_service_instance" "hana_cloud_hana_instance" {
  subaccount_id = resource.btp_subaccount.retest-hana-cloud.id
  # The service plan ID can be looked up via the data source btp_subaccount_service_plan
  serviceplan_id = data.btp_subaccount_service_plan.by_name.id
  name           = "hana-cloud-restest-tf"
  parameters = jsonencode({
    "data" = {
      "memory"                 = 32
      "generateSystemPassword" = true
      "edition"                = "cloud"
    }
  })
  timeouts = {
    create = "25m"
    update = "15m"
    delete = "15m"
  }
}
```

The creation of the setup works as expected. Now changing the value of the `memory` to `64` and executing a `terraform apply` leads to the error:

```bash
Error: API Error Updating Resource Service Instance (Subaccount)
│ 
│   with btp_subaccount_service_instance.hana_cloud_hana_instance,
│   on main.tf line 25, in resource "btp_subaccount_service_instance" "hana_cloud_hana_instance":
│   25: resource "btp_subaccount_service_instance" "hana_cloud_hana_instance" {
│ 
│ json: cannot unmarshal object into Go struct field OperationResponseObject.last_operation.errors of type []servicemanager.ModelError
```

### User's Role Collections

- Global Account Admin
- Subaccount Admin

### Additional context

Even after switching back the memory to 32 GB a `terraform plan` raises the following error when refreshing the state

```terraform
 Error: API Error Reading Resource Service Instance (Subaccount)
│ 
│   with btp_subaccount_service_instance.hana_cloud_hana_instance,
│   on main.tf line 25, in resource "btp_subaccount_service_instance" "hana_cloud_hana_instance":
│   25: resource "btp_subaccount_service_instance" "hana_cloud_hana_instance" {
│ 
│ json: cannot unmarshal object into Go struct field OperationResponseObject.last_operation.errors of type []servicemanager.ModelError
```

This is probably a result of a state change of the resource in the backend to `UPDATE FAILED` which jeopardizes the Terraform state handling

**Comment by lechnerc77** [comment]
The missing field in the service manager response is added in PR #552 including its model based on the API specification. 

However the error is not resolved and needs corrections on SAP HANA Cloud side as the response deviates at several points: in the errors object. Here a sample:

```json
"errors": {
      "error": "BrokerError",
      "description": "Failed update instance request context: map[crm_customer_id: env_type:sapcp global_account_id:XXX instance_name:hana-cloud-restest-tf license_type:SAPDEV origin:sapcp platform:sapcp region:cf-eu12 service_instance_id:XXX signature:GEJm+df7AFKO185tSEibjdgXsdnN44pcqKWp5M67/OpXyXHk+O28GcocqcMQ2Bll+4nBMTwwdWa/dCZhv2ocgMbOKMwbymgxCYW/YCkDX2cT4cOtZRpY9jR+PzJyGZTZmUnso/PN7W9ZP5dYLOaHsuN24JpyRjcLA6PhcFeW8zEcd7jKAunwE8IknkgTroWb+fGziJ/iWXbaLwotjh/KERZlHrCrEFxTN1QJENXcVZs8p9V4w9YiDebk5ed0z12DS7GfUKZ3nUwv3/M4chaYe4zdx1k7YpUqtfENkKCN8krJeXHVHfRkNnulObOBQHUFOU5Un4EqtKRr8Fzwohi4zVeq3r8YfMTHaYyFgqi1mGHfeiT6qG88sszta/3ATnaWnXWFT6Fz3GVwX5FmgWItk1Dn83tWE4eU1muhJPPoxIFkNRnkvVTjL7YpyXmTTCyqigRWPD2Ifz5sjVfkx72dDVPVKBr2xKMyqiobWBI/MCkwgSngH+P8+w3ztih9wUp/8pIulPImLW+EKCPJ+2DGk4+9+vBDiwaRsSkF2qTuyJf/ChMfq/XifXh8KYsULz8Iu+En1YvY6/6QxyzL1w9sPdc3DCfgKwsoIndGf3AXgg39dxgtrjLCwKHfQFfnVO8Kw7YrVhRNG6t6aq/OquQiRgmudOVclefBVh7JN40gg4o= subaccount_id:XXX subdomain:retest-hana-cloud zone_id:XXX], instanceID: XXX, planID: e74f6ed3-8b80-435e-bcdd-ba300754c5dc, serviceID: ab622c22-ad18-4334-b0d5-c6c55da5e6cb, acceptsIncomplete: true: Status: 422; ErrorMessage: \u003cnil\u003e; Description: invalid Parameter (generateSystemPassword): generateSystemPassword cannot be enabled for Update Operations. Please check your parameters and try again. [ref 2YtER5O1L8b6Y7l9rLsGBezlM1Z]; ResponseError: \u003cnil\u003e",
      "broker_error": {
        "StatusCode": 422,
        "ErrorMessage": null,
        "Description": "invalid Parameter (generateSystemPassword): generateSystemPassword cannot be enabled for Update Operations. Please check your parameters and try again. [ref 2YtER5O1L8b6Y7l9rLsGBezlM1Z]",
        "ResponseError": null
      }
    },
 ```

|**Object/Field** | **Deviation** | 
| - | - 
| `errors`  | modelled as  JSON object, but is a JSON array
| `broker_error`  | Field in API description is defined as `brokerError`
| `StatusCode` | Field in API description is defined as  `statusCode`
| `ErrorMessage` | Field in API description is defined as  `errorMessage` 
| `Description` | Field does not exist in API description
| `Operations` | Field does not exist in API description
| `ResponseError` | Field in API description is defined as  `responseError`

**Comment by lechnerc77** [comment]
SAP-internal incident for tracking: <https://jira.tools.sap/browse/HC01-170950>

**Comment by lechnerc77** [comment]
Fix of SM won't do the trick for the provider. Probably need to switch to a generic JSON string for the error response (as already in place for the context)

**Comment by zhjuncai** [comment]
@lechnerc77 may I ask how did you resolve the update error?

**Comment by lechnerc77** [comment]
@zhjuncai : as discussed offline the fix is available with release `1.0.0-rc2` - so if anybody else is running into this issue, please make sure that you upgraded the provider version to 1.0.0-rc2 or later
